app/src/main/java/io/scalac/degree/android/activity/MainActivity.java

```diff
@@ -103,6 +107,60 @@ protected void onStart() {
         loadCoreData();
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        if (toolbarMenuRes != UNKNOWN_MENU_RES) {
+            toolbar.inflateMenu(toolbarMenuRes);
+
+            final MenuItem item = menu.findItem(R.id.action_filter_scheduled);
+            if (settings.filterTalksBySchedule().getOr(false)) {
+                item.setIcon(R.drawable.ic_visibility_white_24dp);
+            } else {
+                item.setIcon(R.drawable.ic_visibility_off_white_24dp);
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if ((item.getItemId() == R.id.action_filter_scheduled)) {
+            final boolean currentState = settings.filterTalksBySchedule().getOr(false);
```

Comment:
To też będzie zmieniane, a jest na zielono bo zmieniałem kolejność metod w klasie :)